### PR TITLE
Dst 298  bugfix

### PR DESCRIPTION
**Pull request by vegard-solum-4ss**
Fixed bug related to modulus and floating point numbers according to DST-298: https://4insight.atlassian.net/jira/software/c/projects/DST/boards/10?modal=detail&selectedIssue=DST-298

### CODE REVIEW

File: src/waveresponse/_core.py [hunk 1]
```diff
@@ -8,6 +8,18 @@
...
+def _robust_modulus(x, periodicity):
+    """
+    Robust modulus operator.
+
+    Similar to ``x % periodicity``, but ensures that it is robust w.r.t. floating
+    point numbers.
+    """
+    x = np.asarray_chkfinite(x % periodicity).copy()
+    np.nextafter(x, -1, where=(x == periodicity), out=x)
+    return x
```
Comment (ace-e4s) [inline src/waveresponse/_core.py:20]:
```suggestion
    return np.nextafter(x, -1, where=(x == periodicity), out=x)
```

Have no other impact then style. Thought it was a bit wierd to have a numpy function dingling there alone.

**Review by ace-e4s — CHANGES_REQUESTED** [review]
Could we add couple of tests to `_robust_modulus` ?

**Comment by vegard-solum-4ss** [comment]
> Could we add couple of tests to `_robust_modulus` ?
Added some tests

**Comment by vegard-solum-4ss** [comment]
Added some tests of `_robust_modulus`

**Review by ace-e4s — APPROVED** [review]

**Pull request merged**